### ISSUE TITLE
Is it possible to use officedown functionalities within thesisdown?

### ISSUE THREAD

**Issue by peranti** [ismayc/thesisdown#205]
_This is not a bug but a query._

--------------

Is it possible to modify the output formats by modifying, similar to the `base_format` argument in `bookdown`? I did not see the same in the `thesisdown` code; please correct me if I overlooked the functionalities of the package.

```
output:
  bookdown::pdf_document2: 
    keep_tex: true
  bookdown::word_document2:
    toc: true
```

FYI: Later on, I would like to import the `officedown` functionalities into the `thesisdown` template.

**Comment by ismayc** [comment]
I don't believe that functionality is currently available. You'd need to dig into the `thesisdown` functions to see if it's possible.

**Comment by peranti** [comment]
Yes, I agree. I couldn't find it possible in the functions as well. Are you aware if this enhancement will be included in the near future? 

Related question. How could we pass a template while interested in a Word document output?

**Comment by ismayc** [comment]
Enhancements to this package are unlikely at this time. I created it 8 years ago and haven’t been in academia for many years. I’m always open to pull requests implementing new functionality, and I’d love to work on this again, but I’ve been saying that for 4 years now…

**Comment by ismayc** [comment]
Good question, and the answer is a genuine limitation rather than an oversight.

`thesisdown::thesis_word()` is a thin wrapper around `bookdown::word_document2(...)`, and that function is defined as:

```r
function (...) markdown_document2(..., base_format = rmarkdown::word_document)
```

It hard-codes `base_format`. So passing your own through the wrapper doesn't work, and R rejects it:

```r
thesisdown::thesis_word(base_format = rmarkdown::word_document)
#> Error: formal argument "base_format" matched by multiple actual arguments
```

I confirmed that error is what you get.

The way through is to skip the wrapper and call bookdown's underlying function directly, which does accept `base_format`:

```yaml
output:
  bookdown::markdown_document2:
    base_format: officedown::rdocx_document
```

That's the officedown plus bookdown route, and it's the same approach officedown documents for bookdown generally. You lose {thesisdown}'s Word wrapper, but that wrapper only sets a couple of knitr chunk options. It has no Word template behind it, so there's little to lose. The Reed LaTeX template is the PDF path only and doesn't carry over to Word in any case.

---

I'm archiving {thesisdown}, so I'm closing out the open issues, and the repository is now read-only. Thank you for taking the time to file this. Reports like yours are what kept the template usable for the people who came after you. If you'd ever like to take over maintenance, I'd be glad to hear from you at chester.ismay@gmail.com.

Chester
